src/rect.lisp

```diff
@@ -236,6 +236,21 @@ value is always a newly allocated SDL_Rect structure."
         (return-from intersect-rect (values nil empty))))
     (values t intersect)))
 
+(defun intersect-rect-and-line (rect x1 y1 x2 y2)
+  "Returns five values where the first value is T if the coordinates of the line
+intersect RECT. The remaining returned values represent the starting and ending
+coordinates of the line clipped to the boundary of the rectangle."
+  (c-with ((x1pos :int) (y1pos :int)
+	   (x2pos :int) (y2pos :int))
+
+    (setf x1pos x1 y1pos y1
```

Comment:
I would argue this is harder to read than each lhs/rhs on its own line.

Reply:
Not a problem. I was originally trying to follow the same style as mouse-state and get-global-mouse-state.